test/unit/db/query-parsers/sql.test.js

```diff
@@ -66,121 +62,102 @@ test('database query parser', function (t) {
-    t.autoend()
-    t.test('should parse a command with gratuitous white space and conditions', function (t) {
+  await t.test('UPDATE SQL', function (t) {
+    t.test('should parse a command with gratuitous white space and conditions', function () {
       const ps = parseSql('  update test set value = 1 where id = 12')
-      t.ok(ps)
+      assert.ok(ps)
 
-      t.ok(ps.operation)
-      t.equal(ps.operation, 'update')
+      assert.ok(ps.operation)
+      assert.equal(ps.operation, 'update')
 
-      t.ok(ps.collection)
-      t.equal(ps.collection, 'test')
-      t.equal(ps.query, 'update test set value = 1 where id = 12')
-      t.end()
+      assert.ok(ps.collection)
+      assert.equal(ps.collection, 'test')
+      assert.equal(ps.query, 'update test set value = 1 where id = 12')
     })
   })
 
-  t.test('INSERT SQL', function (t) {
-    t.autoend()
-    t.test('should parse a command with a subquery', function (t) {
+  await t.test('INSERT SQL', function (t) {
+    t.test('should parse a command with a subquery', function () {
       const ps = parseSql('  insert into\ntest\nselect * from dude')
-      t.ok(ps)
+      assert.ok(ps)
 
-      t.ok(ps.operation)
-      t.equal(ps.operation, 'insert')
+      assert.ok(ps.operation)
+      assert.equal(ps.operation, 'insert')
 
-      t.ok(ps.collection)
-      t.equal(ps.collection, 'test')
-      t.equal(ps.query, 'insert into\ntest\nselect * from dude')
-      t.end()
+      assert.ok(ps.collection)
+      assert.equal(ps.collection, 'test')
+      assert.equal(ps.query, 'insert into\ntest\nselect * from dude')
     })
   })
 
-  t.test('invalid SQL', function (t) {
-    t.autoend()
-    t.test("should return 'other' when handed garbage", function (t) {
+  await t.test('invalid SQL', async (t) => {
+    await t.test("should return 'other' when handed garbage", function () {
       const ps = parseSql('  bulge into\ndudes\nselect * from dude')
-      t.ok(ps)
-      t.equal(ps.operation, 'other')
-      t.notOk(ps.collection)
-      t.equal(ps.query, 'bulge into\ndudes\nselect * from dude')
-      t.end()
+      assert.ok(ps)
+      assert.equal(ps.operation, 'other')
+      assert.ok(!ps.collection)
+      assert.equal(ps.query, 'bulge into\ndudes\nselect * from dude')
     })
 
-    t.test("should return 'other' when handed an object", function (t) {
+    await t.test("should return 'other' when handed an object", function () {
       const ps = parseSql({
         key: 'value'
       })
-      t.ok(ps)
-      t.equal(ps.operation, 'other')
-      t.notOk(ps.collection)
-      t.equal(ps.query, '')
-      t.end()
+      assert.ok(ps)
+      assert.equal(ps.operation, 'other')
+      assert.ok(!ps.collection)
+      assert.equal(ps.query, '')
     })
   })
 
-  t.test('CAT', function (t) {
-    t.autoend()
-    CATs.forEach(function (cat) {
-      t.test(clean(cat.input), function (t) {
-        t.autoend()
+  await t.test('CAT', async function (t) {
```

Comment:
something is getting missed during the back and forth. this is what the test should look like

```js
  await t.test('CAT', async function (t) {
    for (const cat of CATs) {
      await t.test(clean(cat.input), async (t) => {
        const ps = parseSql(cat.input)

        assert.equal(ps.operation, cat.operation, `should parse the operation as ${cat.operation}`)

        if (cat.table === '(subquery)') {
          t.todo('should parse subquery collections as ' + cat.table)
        } else if (/\w+\.\w+/.test(ps.collection)) {
          t.todo('should strip database names from collection names as ' + cat.table)
        } else {
          assert.equal(ps.collection, cat.table, `should parse the collection as ${cat.table}`)
        }
      })
    }
  })
```